hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkInsertOverwriteCommitActionExecutor.java

```diff
@@ -65,9 +70,21 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute() {
 
   @Override
   protected Partitioner getPartitioner(WorkloadProfile profile) {
-    return table.getStorageLayout().layoutPartitionerClass()
-        .map(c -> getLayoutPartitioner(profile, c))
-        .orElse(new SparkInsertOverwritePartitioner(profile, context, table, config));
+    // Fix HUDI-5857, insert overwrite should generate new file group id
```

Comment:
No need to add the JIRA id explicitly because the modifications belong to the current commit.